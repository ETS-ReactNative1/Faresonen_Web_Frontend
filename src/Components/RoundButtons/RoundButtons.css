.round-buttons {
  background: #0a52c1;
  width: 100%;
  display: block;
  margin: 0;
  padding: 10px;
  border-radius: 15px;
  color: white;
  text-align: center;
}

.round-buttons:hover {
  color: white;
  transition: all ease-out 300ms;
  transform: scale(1.1);
}

.round-buttons.finished {
  background: #cf0b0d;
}
