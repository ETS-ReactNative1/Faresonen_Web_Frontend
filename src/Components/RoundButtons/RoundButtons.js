import React, {Component} from "react";
import PropTypes from "prop-types";
import moment from "moment";
import {List} from "semantic-ui-react";
import "./RoundButtons.css";

class RoundButtons extends Component {
  getNextRoundId = (rounds, index) => (rounds[index + 1] ? rounds[index + 1].id : -1);

  createRoundButtons = rounds => rounds.map((round, index, array) => {
    const endDate = round.enddate;

    // eslint-disable-next-line
      round.nextRoundId = this.getNextRoundId(array, index);
    const finished = endDate <= this.props.now;
    const button = this.createRoundButton(round, finished);

    return <List.Item key={round["@uri"]}>{button}</List.Item>;
  });

  createUrl = roundInfo => `#/league/${this.props.tournamentId}/${this.props.seasonId}/${
    this.props.leagueName
  }/round/${roundInfo.roundNo}/roundId/${roundInfo.id}/nextRound/${
    roundInfo.nextRoundId
  }`;

  createFinishedButton = roundInfo => (
    <a href={this.createUrl(roundInfo)} className="round-buttons finished">
      {roundInfo.name}
    </a>
  );

  createNotFinishedButton = roundInfo => (
    <a href={this.createUrl(roundInfo)} className="round-buttons">
      {roundInfo.name}
      <span className="startDate">
        Starter:
        {" "}
        {moment(roundInfo.startdate).format("DD.MM.YYYY")}
      </span>
    </a>
  );

  createRoundButton = (roundInfo, finished = false) => (finished
    ? this.createFinishedButton(roundInfo)
    : this.createNotFinishedButton(roundInfo));

  render() {
    if (!this.props.rounds) {
      return (
        <p>
Kan ikke hente runder for
          {this.props.leagueName}
        </p>
      );
    }
    return <List>{this.createRoundButtons(this.props.rounds)}</List>;
  }
}

RoundButtons.propTypes = {
  leagueName: PropTypes.string.isRequired,
  tournamentId: PropTypes.number.isRequired,
  seasonId: PropTypes.number.isRequired,
  rounds: PropTypes.arrayOf(
    PropTypes.shape({
      "@uri": PropTypes.string.isRequired,
      name: PropTypes.string.isRequired,
      nextRoundId: PropTypes.string,
      roundNo: PropTypes.string.isRequired,
    }),
  ).isRequired,
  now: PropTypes.string.isRequired,
};

export default RoundButtons;
