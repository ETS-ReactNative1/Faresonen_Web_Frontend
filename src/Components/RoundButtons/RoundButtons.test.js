import React from "react";
import {shallow} from "enzyme";
import renderer from "react-test-renderer";
import RoundButtons from "./RoundButtons";

describe("<RoundButtons />", () => {
  it("Should render properly", () => {
    const tree = renderer.create(
      <RoundButtons
        leagueName="Eliteserien"
        tournamentId={1}
        seasonId={340}
        rounds={[
          {
            "@uri": "http://eliteserien.no",
            name: "1. runde",
            nextRoundId: "2",
            roundNo: "1",
            startdate: "2018-04-25T09:33:22.320Z",
          },
        ]}
        now="2018-04-25T09:33:22.320Z"
      />,
    );
    expect(tree).toMatchSnapshot();
  });
});
