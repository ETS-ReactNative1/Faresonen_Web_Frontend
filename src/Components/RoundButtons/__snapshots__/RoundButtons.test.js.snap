// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RoundButtons /> Should render properly 1`] = `
<div
  className="ui list"
  role="list"
>
  <div
    className="item"
    data-value={undefined}
    onClick={[Function]}
    role="listitem"
  >
    <a
      className="round-buttons"
      href="#/league/1/340/Eliteserien/round/1/roundId/undefined/nextRound/-1"
    >
      1. runde
      <span
        className="startDate"
      >
        Starter: 
        30.04.2018
      </span>
    </a>
  </div>
</div>
`;
